free-2.md

```diff
@@ -11,41 +11,81 @@ _What need is met by your idea? Who are you helping? What is that person's
+
+The fluency of the language will likely come from method-chaining rather than function composition because it is easier to see the order in which effects are applied. 
+
+Since the output of programs of this DSL is composites of some sort, there will likely be a designated block of code that puts together those images and texts. Users should also be tasked with defining the location and format of final outputs.
+
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Since this DSL involves two distinct data types, I will assume that it will be built on a statically typed language to help users catch mistakes early. Programming in this DSL will likely occur within the host language’s IDE (instead of a terminal or a separate GUI) to leverage existing tools for code formatting and type checking. 
+
+The high-level order of actions when a program run is as follows:
+- Start at the code block associated with composition (similar to Java’s main function)
+- Check that the given files exist and are of the correct file types
+- Some kind of conversion turns those files into data types that the DSL can more easily operate on
+- Apply the programmed transformations onto the DSL’s representation of a blank canvas
+- Turn that blank canvas into desired output
+
+Error checking is necessary throughout this process because of various potential visual issues every time a new object is placed onto the canvas. Some examples of errors that may occur are overlapping elements and common object-oriented errors like null pointers and out-of-bound indexes. Since this DSL will rely on the IDE, errors will be shown in the terminal. A logging interface within this DSL will also print more human-readable errors in the terminal.
+
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+Things that should be easy to do in this language are compositing images with text and creating multiple images. Possible, but difficult things to do in this language is any sort of text manipulation. The DSL should assume that the set of texts the user uploads at first is mostly final but allow for minor changes like spelling or punctuation errors. One function that should definitely be impossible is writing recursive loops. However, because this idea relies on the host language, I am not exactly sure how I can manually stop recursive loops. Perhaps this is adding some sort of detection in the compiler or a runtime restriction?
+
```

Comment:
Editing text that already exists on an image seems like a very difficult task. Especially since text can come in all sorts of different fonts and colors, and also it can on something already contained within the image.